python/common/org/python/stdlib/datetime/timedelta.java

```diff
@@ -0,0 +1,183 @@
+
+        normalize(days, seconds, microseconds);
+    }
+
+    private void normalize(double days, double seconds, double microseconds) {
+        // convert decimals to smaller units e.g. 1.5 days => 1 days and 43200 seconds
+        seconds += (days % 1) * 60 * 60 * 24;
+        days -= days % 1;
+
+        microseconds += (seconds % 1) * 1e6;
+        seconds -= seconds % 1;
+
+        // round the answer to whole microseconds
+        microseconds = roundHalfToEven(microseconds);
+
+        // convert many microseconds to seconds
+        seconds += (long) (microseconds / 1e6);
+        microseconds -= ((long) (microseconds / 1e6)) * 1e6;
+        // convert many seconds to days
+        days += (long) (seconds / (24 * 60 * 60));
+        seconds -= (long) (seconds / (24 * 60 * 60)) * 24 * 60 * 60;
+
+        if (microseconds < 0) {
+            days -= 1;
+            seconds += MAX_SECONDS;
+            microseconds += MAX_MICROS + 1;
+
+            // IF we overflow MAX_SECONDS again
+            days += (long) (seconds / (24 * 60 * 60));
+            seconds -= (long) (seconds / (24 * 60 * 60)) * 24 * 60 * 60;
+        }
+        if (seconds < 0) {
+            days -= 1;
+            seconds += MAX_SECONDS + 1;
+        }
+
+        if (Math.abs(days) > MAX_DAYS) {
+            throw new org.python.exceptions.OverflowError(
+                    String.format("days=%d; must have magnitude <= %d", (long) days, MAX_DAYS));
+        }
+
+        this.days = org.python.types.Int.getInt((long) days);
+        this.seconds = org.python.types.Int.getInt((long) seconds);
+        this.microseconds = org.python.types.Int.getInt((long) microseconds);
+    }
+
+    private static long roundHalfToEven(double d) {
+        if (d % 0.5 == 0 && d % 1 != 0) {
+            if (Math.floor(d) % 2 == 0) {
+                return (long) Math.floor(d);
+            } else {
+                return (long) Math.ceil(d);
+            }
+        }
+        return Math.round(d);
+    }
+
+    @org.python.Method(__doc__ = "Return the total number of seconds contained in the duration.\n"
+            + " Equivalent to td / timedelta(seconds=1). For interval units \n"
+            + "other than seconds, use the division form directly \n"
+            + "(e.g. td / timedelta(microseconds=1)).", args = {})
+    public org.python.Object total_seconds() {
+        // Regular double precision isn't enough! We need the big stuff
+        BigDecimal daysInSeconds = BigDecimal.valueOf(days.value).multiply(BigDecimal.valueOf(24 * 60 * 60));
+        BigDecimal bigSeconds = BigDecimal.valueOf(seconds.value);
+        BigDecimal microsecondsInSeconds = BigDecimal.valueOf(microseconds.value).divide(BigDecimal.valueOf(1e6));
+        BigDecimal total = daysInSeconds.add(bigSeconds).add(microsecondsInSeconds);
+        return new org.python.types.Float(total.doubleValue());
+    }
+
+    private long getIntValue(org.python.Object obj) {
```

Comment:
Just for reference, this is a bad pattern, but I get why you implemented it: `getIntValue()` should have been implemented in `org.python.types.Int`. The risk is now that each user of `org.python.types.Int` will implement its own getIntValue()

Reply:
Very valid comment, this is very smelly code! Is it worth fixing now, or should we let it be?

Reply:
If you have time it would be great, but it's up to you